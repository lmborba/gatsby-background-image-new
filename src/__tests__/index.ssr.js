/**
 * @jest-environment node
 */

import '@testing-library/react/dont-cleanup-after-each';

import React from 'react';
import ReactDOMServer from 'react-dom/server';
import BackgroundImage from '../index';
import { fixedShapeMock, fluidShapeMock } from './mocks/Various.mock';

describe(`<BackgroundImage /> on SSR`, () => {
  it(`should work within SSR node environment`, () => {
    // Mock Math.random beforehand, lest another random classname is created.
    Math.random = jest.fn(() => 0.424303425546642);
    const renderedBackgroundImage = ReactDOMServer.renderToString(
      <BackgroundImage fluid={fluidShapeMock}>
        <h1>testempty</h1>
      </BackgroundImage>
    );
    expect(renderedBackgroundImage).toMatchSnapshot();
  });

  it(`should work with multiple images in SSR node environment`, () => {
    // Mock Math.random beforehand, lest another random classname is created.
    Math.random = jest.fn(() => 0.424303425546642);
    const renderedBackgroundImage = ReactDOMServer.renderToString(
      <BackgroundImage fixed={[fixedShapeMock, fixedShapeMock]}>
        <h1>testempty</h1>
      </BackgroundImage>
    );
    expect(renderedBackgroundImage).toMatchSnapshot();
  });
});
