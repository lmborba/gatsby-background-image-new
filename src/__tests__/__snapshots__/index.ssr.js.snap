// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackgroundImage /> on SSR should work with multiple images in SSR node environment 1`] = `
"<div class=\\"gbi-1393017994-undefined\\" style=\\"background-position:center;background-repeat:no-repeat;background-size:cover;position:relative;opacity:0.99;display:inline-block\\" data-reactroot=\\"\\"><style>
          .gbi-1393017994-undefined:before,
          .gbi-1393017994-undefined:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            
            transition: opacity 0.5s ease 250ms;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .gbi-1393017994-undefined:before {
            z-index: -100;
            background-image: url(string_of_base64),url(string_of_base64);
            
            opacity: 1; 
          }
          .gbi-1393017994-undefined:after {
            z-index: -101;
            
            
            
          }
        </style><noscript><style>
    .gbi-1393017994-undefined:before {
      opacity: 1;
      background-image: url('test_fixed_image.jpg'),url('test_fixed_image.jpg');
    }</style></noscript><h1>testempty</h1></div>"
`;

exports[`<BackgroundImage /> on SSR should work within SSR node environment 1`] = `
"<div class=\\"gbi-1393017994-undefined\\" style=\\"background-position:center;background-repeat:no-repeat;background-size:cover;position:relative;opacity:0.99\\" data-reactroot=\\"\\"><style>
          .gbi-1393017994-undefined:before,
          .gbi-1393017994-undefined:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            
            transition: opacity 0.5s ease 250ms;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .gbi-1393017994-undefined:before {
            z-index: -100;
            
            
            opacity: 1; 
          }
          .gbi-1393017994-undefined:after {
            z-index: -101;
            
            
            
          }
        </style><noscript><style>
    .gbi-1393017994-undefined:before {
      opacity: 1;
      background-image: url('test_fluid_image.jpg');
    }</style></noscript><h1>testempty</h1></div>"
`;
